src/SystemIO/SystemFolder.cs

```
@@ -104,19 +108,38 @@ public async IAsyncEnumerable<IAddressableStorable> GetItemsAsync(StorableType t
     }
 
     /// <inheritdoc />
-    public Task<IAddressableStorable> GetItemAsync(string id, CancellationToken cancellationToken = default)
+    public Task<IStorableChild> GetItemRecursiveAsync(string id, CancellationToken cancellationToken = default)
     {
+        if (!id.Contains(Path))
+            throw new FileNotFoundException($"The provided ID does not belong to an item in this folder.");
```

Are there any "Storage" related Ids that don't have paths appended? If no maybe just throw "Provided Id is invalid"?

---

It's expected that the provided ID is an ancestor of the current folder, and can always be found if you crawl the object graph for it. The phrasing here reflects that.